content/departments/product-engineering/engineering/cloud/devops/resize-disks-in-statefulset.md

```diff
@@ -0,0 +1,73 @@
+# Resize disks in GKE StatefulSet
+
+Important: this change will caused recreation on all StatefulSet's pods, which can cause donwtime if application is not high-available.
+
+### Steps to resize StatefulSet persistent disks
+
+1. Check the actual disks sizes in GCP
+
+```
+gcloud compute disks list
+```
+
+2. List existing persistent volumes and corresponding persistent volume claims:
+
+```
+kubectl get pv
+kubectl get pvc -n prod
+```
+
+3. Ensure that the storage class of the volumes for resizing has set `AllowVolumeExpansion=True`
+
+```
+kubectl describe sc sourcegraph | grep AllowVolumeExpansion
+# should be: AllowVolumeExpansion:  True
+```
+
+4. Ensure that storage class of the volumes for resizing has set `ReclaimPolicy=Retain`
+
+```
+kubectl describe sc sourcegraph | grep ReclaimPolicy
+# should be: ReclaimPolicy: Retain
+```
+
+5. Modify persistent volume claims to have new size
+
+Note: do it for ALL persistent volumes used by StatefulSet!
+
+```
+kubectl patch pvc repos-gitserver-0 -n prod -p '{ "spec": { "resources": { "requests": { "storage": "250Gi" }}}}'
+```
+
+6. Ensure that all resized persistent volumes, persistent volume claims and google disks has new value (250Gi in this example):
+
+```
+kubectl get pv
+kubectl get pvc -n prod
+gcloud compute disks list
+```
+
+7. Delete StatefulSet without deleting the pods (`--cascade='orphan'`)
+
+Important: without `--cascade='orphan'` you will loose the data!
```

Comment:
```suggestion
Important: without `--cascade='orphan'` you will lose the PVs! (But not the GCP disks since  `Retain` is set on the StorageClass) ```